plugins/background_fetch/background_fetch.cc

```diff
@@ -107,36 +111,17 @@ class BgFetchState
   }
 
 private:
-  BgFetchState() : _log(nullptr), _lock(TSMutexCreate()) {}
-  BgFetchState(BgFetchState const &);   // Don't Implement
-  void operator=(BgFetchState const &); // Don't implement
-
-  TSTextLogObject _log;
   OutstandingRequests _urls;
-  TSMutex _lock;
+  TSTextLogObject _log = nullptr;
+  TSMutex _lock        = TSMutexCreate();
 };
 
 //////////////////////////////////////////////////////////////////////////////
 // Hold and manage some state for the TXN background fetch continuation.
 // This is necessary, because the TXN is likely to not be available
 // during the time we fetch from origin.
 struct BgFetchData {
-  BgFetchData()
-    : hdr_loc(TS_NULL_MLOC),
-      url_loc(TS_NULL_MLOC),
-      vc(nullptr),
-      req_io_buf(nullptr),
-      resp_io_buf(nullptr),
-      req_io_buf_reader(nullptr),
-      resp_io_buf_reader(nullptr),
-      r_vio(nullptr),
-      w_vio(nullptr),
-      _bytes(0),
-      _cont(nullptr)
-  {
-    mbuf = TSMBufferCreate();
-    memset(&client_ip, 0, sizeof(client_ip));
-  }
+  BgFetchData() { memset(&client_ip, 0, sizeof(client_ip)); }
```

Comment:
`ink_zero(client_ip)`?

Reply:
I don't see any other plugin pulling this in... We can do so later I guess, but I'm still torn on this integration with "internals" into plugins. The downside is that if we do, then "forking" an ATS plugin and build it outside of ATS (with tsxs) gets a lot more complicated :-/. I really kinda think we should have avoided using anything from the core in plugins, until those APIs are exposed publicly.